python/tests/test_run_config.py

```diff
@@ -42,3 +44,22 @@ async def test_load_parametrized_data_config(set_temporary_env_vars):
                 },
             ],
         }
+
+@pytest.mark.asyncio
+async def test_running_prompt_with_dependencies(set_temporary_env_vars):
+    """Test running a prompt with dependencies with the run_with_dependencies flag set to True"""
+
+    mock_openai = Mock(side_effect=mock_openai_chat_completion_with_dependencies)
+
+    with patch.object(openai.chat.completions, "create", new=mock_openai):
+        config_relative_path = "aiconfigs/tarvel_gpt_prompts_with_dependency.json"
+        config_absolute_path = get_absolute_file_path_from_relative(__file__, config_relative_path)
+        config = AIConfigRuntime.load(config_absolute_path)
+
+        combined_prompt_parameters = {
+            "city": "NYC",
+            "order_by": "geographic location",
+        }
+        await config.run("gen_itinerary", combined_prompt_parameters, run_with_dependencies=True)
+
+    assert mock_openai.call_count == 2
```

Comment:
Just checking that you're rebased onto main, specifically after #999?

Reply:
Haven't rebased yet.

Reply:
I did notice tests failing after a merge. I guess #999 was the case.
I reverted the merge later.

Reply:
Yep following the changes in PR #999, its necessary to revise our approach to mocking the OpenAI client. Your proposed code looks solid to me. My suggestions are:

1. Rebase your diff to main
2. Temporarily mark this test as an expected failure. Please add the @pytest.mark.xfail decorator above the existing @pytest.mark.asyncio decorator."

Reply:
Will do so!

Reply:
Done!